packages/serverpod/lib/src/util/path_util.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:path/path.dart' as p;
+
+/// Provides utility functions for working with paths.
+class PathUtil {
+  /// Tests if a file is in a whitelisted path. File paths should all be
+  /// relative. Complete directories can be whitelisted by adding a trailing
+  /// slash. The path and whitelisted paths are case sensitive and should be
+  /// in unix/posix format.
+  static bool isFileWhitelisted(String path, Set<String> whitelistedFiles) {
+    for (var whitelistedFile in whitelistedFiles) {
+      if (whitelistedFile.endsWith('/')) {
+        if (path.startsWith(whitelistedFile)) {
```

Comment:
Question: Wouldn't this make a trailing `/` whitelist all possible subfolders of that folder? Is that intentional?

I.e. if `path/to/ ` was whitelisted then it would be possible to retrieve `path/to/not/this/file.txt` as well?

Reply:
This is intended to work the same way as paths in the Flutter pubspec. Will clarify in documentation.